### ISSUE TITLE
Investigate why tests don't fail when the app become unresponsive

### ISSUE THREAD

**Issue by Valentina1133** [status-im/status-desktop#15249]
### Description

https://ci.status.im/job/status-desktop/job/e2e/job/manual/2113/ example

We faced the issue, when the app becomes unresponsive, test don't fail but everything is just freezing forever (until timeout expires)

First thoughts - add timeout to each test, so in such case it will fail at least after timeout expires

**Comment by Valentina1133** [comment]
For now I managed to set it locally.

CI - faced the issue. 
+ python3 -m pytest --reruns=1 -k=test_add_edit_and_remove_permissions --log-level=INFO --log-cli-level=INFO --disable-warnings --alluredir=./allure-results --timeout=240 flag is only set when I put the exact name of test to TEST_NAME field, when I put permissions to TEST_NAME it is not set for some reason !

Continue investigation

**Comment by Valentina1133** [comment]
https://ci.status.im/job/status-desktop/job/e2e/job/manual/2127/console - looks like flag was added to command line but for some reason still didn't work when running all permissions tests

**Comment by Valentina1133** [comment]
https://github.com/status-im/desktop-qa-automation/issues/567 - will try once again to implement evert handler thing tomorrow

**Comment by Valentina1133** [comment]
The best option here after I investigated and tried several things:
setting global timeout with pytest-timeout. So when it's set globally, each test would fail in case test execution is more than set timeout.

The problem was to combine it with rerun. It turned out that they are incompatible. So I was looking for a way to combine it. After investigation 2 options were found and tested. The results:
1) --rerun-except Timeout (but in this case after rerun there is no timeout anymore)
2)-o timeout_func_only=true - in this case timeout set for both attempts

So I chose 2d option. It works fine, tested on the build with permissions bug - https://ci.status.im/job/status-desktop/job/e2e/job/manual/2164/allure/#suites/ea35bd66d78deeec4a096260dd1d8144/991d57c78fc381bd
